providers/implementations/ciphers/cipher_aes_gcm_hw_ppc.inc

```diff
@@ -23,6 +23,114 @@ static int aes_ppc_gcm_initkey(PROV_GCM_CTX *ctx, const unsigned char *key,
     return 1;
 }
 
+
+extern size_t ppc_aes_gcm_encrypt(const unsigned char *in, unsigned char *out, size_t len,
+                         const void *key, unsigned char ivec[16], u64 *Xi);
+extern size_t ppc_aes_gcm_decrypt(const unsigned char *in, unsigned char *out, size_t len,
+                         const void *key, unsigned char ivec[16], u64 *Xi);
+
+static inline int IsBigEndian(int i)
+{
+    return ! *((char *)&i);
+}
```

Comment:
I am really confused why this different handling for different endianness is needed. If I understand the calls right, it seems like you're applying the BE/LE conversion to the counter in IVEC, but that counter should be always in the same endianness, otherwise there would be a mismatch between the BE/LE systems in what the encrypted data is.